mailpoet/tests/integration/Segments/DynamicSegments/Filters/SubscriberDateFieldTest.php

```diff
@@ -0,0 +1,205 @@
+      ->create();
+    (new Subscriber())
+      ->withEmail('3@example.com')
+      ->withLastPurchaseAt(new CarbonImmutable('2023-07-13'))
+      ->create();
+    (new Subscriber())
+      ->withEmail('4@example.com')
+      ->withLastPurchaseAt(new CarbonImmutable('2023-07-14'))
+      ->create();
+    (new Subscriber())
+      ->withEmail('5@example.com')
+      ->withLastPurchaseAt(new CarbonImmutable('2023-07-15'))
+      ->create();
+    $this->assertFilterReturnsEmails('lastPurchaseDate', 'after', '2023-07-13', ['4@example.com', '5@example.com']);
+  }
+
+  public function testItWorksForOn(): void {
+    (new Subscriber())
+      ->withEmail('1@example.com')
+      ->withLastClickAt(new CarbonImmutable('2023-07-11'))
+      ->create();
+    (new Subscriber())
+      ->withEmail('2@example.com')
+      ->withLastClickAt(new CarbonImmutable('2023-07-12'))
+      ->create();
+    (new Subscriber())
+      ->withEmail('3@example.com')
+      ->withLastClickAt(new CarbonImmutable('2023-07-13'))
+      ->create();
+    $this->assertFilterReturnsEmails('lastClickDate', 'on', '2023-07-12', ['2@example.com']);
+  }
+
+  public function testItWorksForOnOrBefore(): void {
+    (new Subscriber())
+      ->withEmail('1@example.com')
+      ->withLastOpenAt(new CarbonImmutable('2023-07-11'))
+      ->create();
+    (new Subscriber())
+      ->withEmail('2@example.com')
+      ->withLastOpenAt(new CarbonImmutable('2023-07-12'))
+      ->create();
+    (new Subscriber())
+      ->withEmail('3@example.com')
+      ->withLastOpenAt(new CarbonImmutable('2023-07-13'))
+      ->create();
+    (new Subscriber())
+      ->withEmail('4@example.com')
+      ->withLastOpenAt(new CarbonImmutable('2023-07-14'))
+      ->create();
+    (new Subscriber())
+      ->withEmail('5@example.com')
+      ->withLastOpenAt(new CarbonImmutable('2023-07-15'))
+      ->create();
+    $this->assertFilterReturnsEmails('lastOpenDate', 'onOrBefore', '2023-07-13', ['1@example.com', '2@example.com', '3@example.com']);
+  }
+
+  public function testItWorksForOnOrAfter(): void {
+    (new Subscriber())
+      ->withEmail('1@example.com')
+      ->withLastPageViewAt(new CarbonImmutable('2023-07-11'))
+      ->create();
+    (new Subscriber())
+      ->withEmail('2@example.com')
+      ->withLastPageViewAt(new CarbonImmutable('2023-07-12'))
+      ->create();
+    (new Subscriber())
+      ->withEmail('3@example.com')
+      ->withLastPageViewAt(new CarbonImmutable('2023-07-13'))
+      ->create();
+    (new Subscriber())
+      ->withEmail('4@example.com')
+      ->withLastPageViewAt(new CarbonImmutable('2023-07-14'))
+      ->create();
+    (new Subscriber())
+      ->withEmail('5@example.com')
+      ->withLastPageViewAt(new CarbonImmutable('2023-07-15'))
+      ->create();
+    $this->assertFilterReturnsEmails('lastPageViewDate', 'onOrAfter', '2023-07-13', ['3@example.com','4@example.com', '5@example.com']);
+  }
+
+  public function testItWorksForNotOn(): void {
```

Comment:
> For the tests, I opted for using different fields for the different operators. I could have written tests for every combination of action/operator, but that seemed excessive, and I couldn't immediately think of an elegant way to handle it with a dataProvider method. I would welcome other ideas for how to handle the tests.

I thought about it and your approach is something I would do. I was also considering something more complex, but it could be very complex, which means easier bugs in the code.